server/server.go

```diff
@@ -2038,12 +2039,27 @@ func (s *Server) getCompiler() *ast.Compiler {
 
 func (s *Server) makeRego(ctx context.Context, partial bool, txn storage.Transaction, input ast.Value, path string, m metrics.Metrics, instrument bool, tracer topdown.Tracer, opts []func(*rego.Rego)) (*rego.Rego, error) {
 
+	opts = append(
+		opts,
+		rego.Transaction(txn),
+		rego.Query(path),
+		rego.ParsedInput(input),
+		rego.Metrics(m),
+		rego.Tracer(tracer),
+		rego.Instrument(instrument),
+		rego.Runtime(s.runtime),
+		rego.UnsafeBuiltins(unsafeBuiltinsMap),
+	)
+
 	if partial {
+		// pick a namespace for the query (path), doesn't really matter what it is
+		// as long as it is unique for each path.
+		namespace := fmt.Sprintf("partial_%d", xxhash.ChecksumString64(path))
```

Comment:
What happens if we use the path as the partial namespace without hashing it?

Reply:
We can, we just need to transform it into something that is a valid `Var` string. I was less concerned about collisions since I think in practice people don't use a huge variety of queries (this issue itself was only noticed while reading through the code debugging something else).

I'll take a look at just converting them into safe var strings from the path. I agree that its the safest option.

Reply:
Can the namespace just be something like `partial["<path>"]`?

Reply:
💡 probably, yea

Its used as the module ID, which can be any string, and as part of the package name so that should be OK (we can just like "%q" the path or something when sticking it in there)

Reply:
Latest version just uses the url path, and I relaxed the check on the partial namespace so as long as its a valid package name its OK.